app/models/miq_cockpit_ws_worker/runner.rb

```diff
@@ -0,0 +1,65 @@
+class MiqCockpitWsWorker::Runner < MiqWorker::Runner
+  BINDING_ADDRESS = ENV['BINDING_ADDRESS'] || (Rails.env.production? ? nil : "0.0.0.0")
+
+  def self.start_worker(*args)
+    runner = MiqCockpitWsWorker::Runner.new(*args)
+
+    runner.prepare
+    runner.start_cockpit_ws
+    runner.run
+  end
+
+  def do_work
+  end
+
+  def pid_file
+    Rails.root.join("tmp/pids/cockpit-ws.pid")
+  end
+
+  def delete_pid_file
+    File.delete(pid_file) if File.exist?(pid_file)
+  end
+
+  def heartbeat(*args)
+    ret = super
+    if !@cockpit_pid || !MiqProcess.alive?(@cockpit_pid)
+      @cockpit_pid = nil
+      do_exit("Cockpit-ws not running, exiting!", 1)
+    end
+    ret
+  end
+
+  def clean_up_running_process(pid)
+    return unless pid && pid > 2
+    begin
+      _log.info("Terminating cockpit_ws")
+      Process.kill("TERM", pid)
+    rescue Errno::ESRCH
+      _log.warn("cockpit-ws has been killed")
+    rescue => err
+      _log.warn("cockpit-ws has been killed, but with the following error: #{err}")
+    end
+  end
+
+  def before_exit(_message, _exit_code)
+    clean_up_running_process(@cockpit_pid)
+    delete_pid_file
+  end
+
+  def start_cockpit_ws
+    if File.exist?(pid_file)
+      clean_up_running_process(File.read(pid_file).to_i)
+      delete_pid_file
+    end
+
+    opts = @worker_settings
+    cockpit_ws = MiqCockpit::WS.new(opts)
+    cockpit_ws.save_config
+    pid = Kernel.spawn({"XDG_CONFIG_DIRS"     => cockpit_ws.config_dir,
+                        "MIQ_WEB_SERVICE_URI" => MiqCockpitWsWorker.web_service_uri},
+                       cockpit_ws.command(BINDING_ADDRESS))
+    Process.detach(pid)
```

Comment:
Do you want signals to be sent to the child process?  That is if we send a KILL or a TERM to the worker, should that be sent to the child process as well?